File: editor/app/src/main/java/fbla/editor/animationStates.java
package fbla.editor;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

import com.google.gson.annotations.SerializedName;

public class animationStates {
    @SerializedName("idle")
    private List<String> idleImages;
    @SerializedName("walking_up")
    private List<String> walkingUpImages;
    @SerializedName("walking_down")
    private List<String> walkingDownImages;
    @SerializedName("walking_left")
    private List<String> walkingLeftImages;
    @SerializedName("walking_right")
    private List<String> walkingRightImages;

    // does not matter what setallempty is, it has to be there so gson doesnt try to use it
    public animationStates(boolean setAllEmpty){
        this.idleImages = new ArrayList<>();
        this.walkingDownImages = new ArrayList<>();
        this.walkingUpImages = new ArrayList<>();
        this.walkingLeftImages = new ArrayList<>();
        this.walkingRightImages = new ArrayList<>();
    }

    public List<String> getIdleImagesPaths(){
        return idleImages;
    }
    public List<String> getWalkingUpImagesPaths(){
        return walkingUpImages;
    }
    public List<String> getWalkingDownImagesPaths(){
        return walkingDownImages;
    }
    public List<String> getWalkingLeftImagesPaths(){
        return walkingLeftImages;
    }
    public List<String> getWalkingRightImagesPaths(){
        return walkingRightImages;
    }
}

File: editor/html-rewrite/js/AnimationStates.js
class AnimationStates {
    idle = [];
    walking_up = [];
    walking_down = [];
    walking_left = [];
    walking_right = [];
    constructor (idle, walking_up, walking_down, walking_left, walking_right){
        this.idle = idle;
        this.walking_up = walking_up;
        this.walking_down = walking_down;
        this.walking_left = walking_left;
        this.walking_right = walking_right;
    }
    get getidle(){
        return this.idle;
    }
    get getwalking_up(){
        return this.walking_up;
    }
    get getwalking_down(){
        return this.walking_down;
    }
    get getwalking_left(){
        return this.walking_left;
    }
    get getwalking_right(){
        return this.walking_right;
    }
}